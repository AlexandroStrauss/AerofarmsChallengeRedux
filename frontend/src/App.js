import React, { Suspense } from 'react';
import MasterSensor from './components/MasterSensorChart';
import SidebarHook from './components/SidebarHook';

function App() {
  // const { i18n } = useTranslation();

  // function ChangeLanguage(lang) {
  //   i18n.changeLanguage(lang);
  // }


  return (
    // <Suspense fallback="spinner">
      <div className="main-container">

      {/* <NavBar /> */}
      <Suspense fallback = "loading"> 
        <SidebarHook />
      </Suspense>

      {/* <Button className="transButton" onClick={()=>ChangeLanguage('pt')} > 
          Portugues
      </Button>

      <Button className="transButton" onClick={()=>ChangeLanguage('en')} > 
          English
      </Button>
 */}
        {/* <Card style={{ width: '18rem' }}>
          <Card.Body>
            <Card.Title>Card Title</Card.Title>
            <Card.Text>
              Some quick example text to build on the card title and make up the bulk of
              the card's content.
            </Card.Text>
            <Button variant="primary">Go somewhere</Button>
          </Card.Body>
        </Card> */}

        {/* I've wrapped both chart components in larger "Master" components. This makes it easier to manage the button inputs I want to have.
        The buttons affect the variables passed into the query in the chart components, and thus what is displayed on the chart. 
        Having them adjust state in a higher-order component makes it easier to control what's passed down to the chart and its query at any point.
        
        Since the two "Master" and the two chart components are functionally very similar, I've included further comments only in the "Temperature" version of each.
        */}
        {/* <div id="component-container">

        <div id="graph-title">Temperature By Room</div>
            <MasterTemperatureBySensor />
        </div>

        <div id="component-container">

        <div id="graph-title">Humidity By Room</div>
            <MasterHumidityBySensor />
        </div> */}

        <div id="component-container">
          <Suspense fallback="loading">
          <MasterSensor />
          </Suspense>
        </div>

      </div>
    // </Suspense>
  );
}

export default App;
