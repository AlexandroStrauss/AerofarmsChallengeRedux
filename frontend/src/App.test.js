import React, { Suspense } from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import wait from 'waait';
import { I18nextProvider } from 'react-i18next';
import i18n from '../src/i18n';
import 'regenerator-runtime/runtime';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';

Enzyme.configure({ adapter: new Adapter() })
it('renders without crashing', async () => {

  let tree = renderer.create(

    <MockedProvider addTypename={false}>
      {/* <Suspense fallback="loading"> */}
        {/* <I18nextProvider i18n={i18n}> */}
          <App />
        {/* </I18nextProvider> */}
      {/* </Suspense> */}
    </MockedProvider>
    ).toJSON();


    // let tree = Enzyme.mount(
    //   // <I18nextProvider i18n={i18n}>
    //   <App />
    //   // </I18nextProvider>
    // )
  
  // })

  await renderer.act(async () => {
    await wait(0); // wait for response

  });

    
  console.log(tree);
  expect(tree).toMatchSnapshot();
});


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
