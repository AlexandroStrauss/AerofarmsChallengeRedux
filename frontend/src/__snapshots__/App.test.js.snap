// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="main-container"
>
  loading
  <div
    id="component-container"
  >
    loading
  </div>
</div>
`;
